pkg/generators/listergen/listergen.go

```diff
@@ -104,112 +119,83 @@ func (g Generator) Run(ctx *genall.GenerationContext, f flag.Flags) error {
 }
 
 func (g *Generator) setDefaults(f flag.Flags) (err error) {
-	if f.InputDir != "" {
-		g.inputDir = f.InputDir
-		pkg, hasGoMod := util.CurrentPackage(f.InputDir)
-		if len(pkg) == 0 {
-			return fmt.Errorf("error finding the module path for this package %q", f.InputDir)
-		}
-		g.inputpkgPaths = pkgPaths{
-			basePackage: pkg,
-			hasGoMod:    hasGoMod,
-		}
+	g.inputDir = f.InputDir
+	absoluteInputDir, err := filepath.Abs(g.inputDir)
```

Comment:
Not sure if its already been tested, but there was an issue with using absolute path when APIs were defined in a sub-module and the generator was run from a parent directory. That was the reason it was reverted [here](https://github.com/kcp-dev/code-generator/pull/30/files). But please feel free to ignore it if it behaves as required.

Reply:
must have copied it to informergen before the reversion, changed it for lister and informergen